src/main/java/seedu/address/logic/parser/NumberParser.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+
+public class NumberParser {
+
+    private final int index;
+    private final Command lastCommand;
+
+    /**
+     * Parses number inputs. Used for confirming which clients the user is referring to
+     * in edit and delete scenarios with clients with similar names
+     *
```

Comment:
Maybe this info can be used as class Javadoc? Then here just state that this is a constructor. 
Maybe you can consider rewriting the second statement as: 
`Used for confirming which clients the user is referring to in edit and delete scenarios where clients have similar names`